client/components/Signup.js

```diff
@@ -0,0 +1,99 @@
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
+import { signup } from '../reducers'
+
+
+export class Signup extends Component {
+  constructor() {
+    super()
+    this.state = {
+      firstNameInput: '',
+      lastNameInput: '',
+      emailInput: '',
+      passwordInput: '',
+      dirty: false
```

Comment:
what does `dirty` do in the state?

Reply:
it just tracks whether you have typed anything in the form yet